server/protocols/webadmin/webadmin-data/src/test/java/org/apache/james/webadmin/routes/DLPRulesRoutesTest.java

```diff
@@ -0,0 +1,410 @@
+                .put(DEFAULT_DOMAIN)
+            .then()
+            .log().all()
+                .statusCode(HttpStatus.NO_CONTENT_204);
+
+            assertThat(dlpStore.list(SENDER_DOMAIN))
+                .hasSize(2)
+                .extracting(dlpConfiguration -> dlpConfiguration.getId().asString())
+                .containsExactlyInAnyOrder("1", "2");
+        }
+
+        @Test
+        void putShouldstoreTheConfigurationsWhenTargetsAreNotSpecified() {
+            String body = JsonNodeFactory.instance.arrayNode()
+                .add(JSON_WITHOUT_TARGET)
+                .toString();
+
+            given()
+                .body(body)
+            .when()
+                .put(DEFAULT_DOMAIN)
+            .then()
+                .statusCode(HttpStatus.NO_CONTENT_204);
+
+            assertThat(dlpStore.list(SENDER_DOMAIN))
+                .hasSize(1)
+                .extracting(dlpConfiguration -> dlpConfiguration.getId().asString())
+                .containsOnly("3");
+        }
+
+        @Test
+        void putShouldReturnBadRequestWhenIdIsNotSpecified() {
+            String body = JsonNodeFactory.instance.arrayNode()
+                .add(JSON_WITHOUT_ID)
+                .toString();
+
+            given()
+                .body(body)
+            .when()
+                .put(DEFAULT_DOMAIN)
+            .then()
+                .statusCode(HttpStatus.BAD_REQUEST_400)
+                .contentType(JSON_CONTENT_TYPE)
+                .body("statusCode", is(400))
+                .body("type", is("InvalidArgument"))
+                .body("message", is("id field is mandatory"));
+        }
+
+        @Test
+        void putShouldReturnBadRequestWhenExpressionIsNotSpecified() {
+            String body = JsonNodeFactory.instance.arrayNode()
+                .add(JSON_WITHOUT_EXPRESSION)
+                .toString();
+
+            given()
+                .body(body)
+            .when()
+                .put(DEFAULT_DOMAIN)
+            .then()
+                .statusCode(HttpStatus.BAD_REQUEST_400)
+                .contentType(JSON_CONTENT_TYPE)
+                .body("statusCode", is(400))
+                .body("type", is("InvalidArgument"))
+                .body("message", is("expression field is mandatory"));
+        }
+
+        @Test
+        void putShouldReturnNotFoundWhenDomainNotInList() {
+            String body = JsonNodeFactory.instance.arrayNode()
+                .add(JSON_WITH_POSITIVE_TARGETS)
+                .add(JSON_WITH_NEGATIVE_TARGETS)
+                .toString();
+
+            given()
+                .body(body)
+            .when()
+                .put("strange.com")
+            .then()
+                .statusCode(HttpStatus.NOT_FOUND_404)
+                .contentType(JSON_CONTENT_TYPE)
+                .body("statusCode", is(404))
+                .body("type", is("InvalidArgument"))
+                .body("message", is("The domain list does not contain: strange.com"));
```

Comment:
DomainList is some specific James components and wording tht might be criptic for admins. 

I would suggest: `'strange.com' is not managed by this James server.`